docs/manual/scala/guide/cluster/DomainModelling.md

```diff
@@ -1,19 +1,121 @@
+
+Lagom embraces Event Sourcing and CQRS by promoting the usage of Akka Persistence Typed in combination the CQRS principle. Akka Persistence Typed provides an API for building event sourced actors, but not necessarily for CQRS Aggregates. To build CQRS applications we need to apply a few rules to our design.
+
+We will explain the API and concepts while building a simplified shopping cart model. A full-fledge shopping cart sample can be found on our [samples repository](https://github.com/lagom/lagom-samples/tree/1.6.x/shopping-cart/shopping-cart-scala).
+
+You will learn, how to:
+
+* model a CQRS Aggregate using Akka Persistence Typed
+* prepare its event journal to be queryable using event tags
+* register the model on Akka's Cluster Sharding
+* look up instances on the cluster and iteract with them
+* define serializers for your commands, events and model state
+* understand the main configuation settings and their raison-être
 
 ## Encoding the model
-  * typed protocol - Command, Event and State
-  * defining typed replies and error handling
-  * Enforced replies
-  * Changing behavior - FSM
+
+ We will start by defining the rules of our shopping cart.
+
+* a shopping cart starts empty, gets items added to it and can be either cancelled or checkedout
+* an empty shopping cart can't be checkedout
+* a cancelled or checkedout can't have new items added to it
+* items are simplified to a `name` and a `quantity`
+* when you re-add an item, you override the exisitng by change its quantity
+* you delete an item by updating its quantity to 0
+
+### Typed protocol - Command, Event and State
+
+We will start by define our model in terms of Commands, Events and State.
+
+The state of our shopping cart will be define as following:
+
+```scala
+/** Common Shopping Cart trait */
+sealed trait ShoppingCart
+/** A shopping cart starts as empty */
+case object EmptyShoppingCart extends ShoppingCart
+/** Once we add new items it becomes an open shopping cart */
+case class OpenShoppingCart(items: Map[String, Int]) extends ShoppingCart
+/** Once cancelled it reaches end-of-life and can't be used anymore */
+case object CancelledShoppingCart extends ShoppingCart
+/** Once checked-out it reaches end-of-life and can't be used anymore */
+case object CheckedOutShoppingCart extends ShoppingCart
+```
+
+Next we define the commands that we can send to it. Each command has a `replyTo:ActorRef[R]` field where `R` is the reply that will be sent back to the caller. We will cover it in detail in a while.
+
+```scala
+// Replies
+sealed trait Confirmation
+sealed trait Accepted extends Confirmation
+case object Accepted extends Accepted
+final case class Rejected(reason: String) extends Confirmation
+final case class CurrentState(items: Map[String, Int]) extends ShoppingCartReply
```

Comment:
`ShoppingCartReply` is not in the snippet. A left-over?

Reply:
(nitpick) 
I just realised (probably that only happens on this code) that `case class CurrentState` doesn't contain the current state but the `items`. What will that containt when the state is `cancelled`? Could we rename `case class CurrentState` to something else?

Reply:
> ShoppingCartReply is not in the snippet. A left-over?

`ShoppingCartReply` is not needed anymore. This was a requirement when we needed to extend `ExpectedReply`. That's remove from Akka now.

> What will that containt when the state is cancelled? 

You are right about the state. That's an oversight. I will review it. What do you think about a simplified state class defined as `Cart(items: Map, status: String)`? 

> Could we rename case class CurrentState to something else?

Suggestions?

Reply:
@ignasi35, I realise now that I forgot to remove the last reference to `ShoppingCartReply` hence your comment. I will remove it. Not needed anymore as mentioned.

Reply:
I'm experimenting with `final case class CartState(items: Map[String, Int], status: String)`

and an explanation in the guide saying: "`CartState` is not the shopping-cart itself, but the representation we want to expose to the external world"